--!strict
local RunService = game:GetService("RunService")
local platform_icon = if RunService:IsServer() then "[ğŸ—¼]" else "[ğŸ’»]"

local types = require(script.Parent.types)

--- Roblox logging sink.
local roblox: types.sink = function(log)
    if log.type == "debug" then
        print(`[{platform_icon}][ğŸ•’{log.timestamp.clock}][ğŸ”¨]: {log.msg}`)
    elseif log.type == "info" then
        print(`[{platform_icon}][ğŸ•’{log.timestamp.clock}][â„¹ï¸]: {log.msg}`)
    elseif log.type == "warn" then
        warn(`[{platform_icon}][ğŸ•’{log.timestamp.clock}][âš ï¸]: {log.msg}`)
    elseif log.type == "error" then
        warn(`[{platform_icon}][ğŸ•’{log.timestamp.clock}][âœ–ï¸]: {log.msg}\n{log.trace}`)
    elseif log.type == "fatal" then
        task.spawn(error, `[{platform_icon}][ğŸ•’{log.timestamp.clock}][âŒ]: {log.msg}\n{log.trace}`, 0)
    end
end

return {
    roblox = roblox,
}
